models/ed/R/met2model.ED2.R

```diff
@@ -274,31 +277,33 @@ met2model.ED2 <- function(in.path, in.prefix, outfolder, start_date, end_date, l
     }
 
     ## write DRIVER file
-    sites <- 1
-    metgrid <- c(1, 1, 1, 1, lon, lat)
     metvar <- c("nbdsf", "nddsf", "vbdsf", "vddsf", "prate", "dlwrf",
                 "pres", "hgt", "ugrd", "vgrd", "sh", "tmp", "co2")
-    nmet <- length(metvar)
-    metfrq <- rep(dt, nmet)
-    metflag <- rep(1, nmet)
+    metvar_table <- data.frame(
+      variable = metvar,
+      update_frequency = dt,
+      flag = 1
+    )
     if (!useCO2) {
-      metflag[metvar == "co2"] <- 4
-      metfrq[metvar == "co2"] <- 380
+      metvar_table[metvar_table$variable == "co2",
+                   c("update_frequency", "flag")] <- list(380, 4)
     }
-    write.table("header", met_header, row.names = FALSE, col.names = FALSE)
-    write.table(sites, met_header, row.names = FALSE, col.names = FALSE, append = TRUE)
-    write.table(met_folder, met_header, row.names = FALSE, col.names = FALSE, append = TRUE,
-                quote = FALSE)
-    write.table(matrix(metgrid, nrow = 1), met_header, row.names = FALSE, col.names = FALSE,
-                append = TRUE, quote = FALSE)
-    write.table(nmet, met_header, row.names = FALSE, col.names = FALSE, append = TRUE, quote = FALSE)
-    write.table(matrix(metvar, nrow = 1), met_header, row.names = FALSE, col.names = FALSE, append = TRUE)
-    write.table(matrix(metfrq, nrow = 1), met_header, row.names = FALSE, col.names = FALSE, append = TRUE,
-                quote = FALSE)
-    write.table(matrix(metflag, nrow = 1), met_header, row.names = FALSE, col.names = FALSE,
-                append = TRUE, quote = FALSE)
+
+    ed_metheader <- list(list(
+      path_prefix = met_folder,
+      nlon = 1,
+      nlat = 1,
+      dx = 1,
+      dy = 1,
+      xmin = lon,
+      ymin = lat,
+      variables = metvar_table
+    ))
+
+    check_ed_metheader(ed_metheader)
+    write_ed_metheader(ed_metheader, met_header)
```

Comment:
now these two object names are two similar, maybe we can call the latter `met_header_file` or something?

Reply:
Addressed in 7857f5c.